Add RemoveMaxCorrelation TabularTransform

floscha opened the pull request:
This PR adds a new TabularTransform class to automatically remove features that correlate with the target variable more than a given threshold (0.95 by default). The PR is heavily inspired by TransmogrifAI's [SanityChecker](https://github.com/salesforce/TransmogrifAI/blob/master/core/src/main/scala/com/salesforce/op/stages/impl/preparators/SanityChecker.scala) and includes the required unit tests.

jph00 commented:
Thanks - I think this seems like a useful thing to have. You've got quite a bit of duplicate code in there - could you please try to make this more concise? Also, see if you can make it a bit more consistent with our style guide: https://docs-dev.fast.ai/style.html . And add some docstrings (see other fastai modules for examples).

In a future PR perhaps you could also add docs for your class (fastai_docs repo)?

sgugger commented:
Please amend this PR to work with the latest big refactor and adapt to the comment above. I'm closing this for now, thanks

floscha commented:
Hi @jph00 and @sgugger, I rebased this PR on the latest refactor and addressed all comments (removed duplicate code, adhere to the style guide and added docstrings). If hope these changes are sufficient for the PR to be merged into the master. Then I will also make sure to add some documentation to the fastai_docs repo.
